.github/workflows/pipeline.yml

```diff
@@ -31,65 +31,93 @@ env:
 
 jobs:
   #---------------------------------------------------------------------
-  # check_krnlmon_dpdk
+  # 1 - dpdk_build_check
   #---------------------------------------------------------------------
-  check_krnlmon_dpdk:
+  dpdk_build_check:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Check out krnlmon repository
+      - name: Check out networking-recipe
         uses: actions/checkout@v3
         with:
-          path: krnlmon
+          repository: ipdk-io/networking-recipe
+          submodules: recursive
+          path: recipe
 
-      - name: Check out SAI repository
-        uses: actions/checkout@v3
+      - name: Install dpdk-sde
+        uses: robinraju/release-downloader@v1.8
         with:
-          repository: opencomputeproject/SAI
-          path: SAI
+          repository: ${{ env.SDE_REPOSITORY }}
+          tag: ${{ env.SDE_TAG }}
+          fileName: ${{ env.SDE_FILENAME }}
+      - run: |
+          sudo tar -xzf $SDE_FILENAME -C /
+          rm $SDE_FILENAME
 
-      - name: Check out networking-recipe
-        uses: actions/checkout@v3
+      - name: Install stratum-deps
+        uses: robinraju/release-downloader@v1.8
         with:
-          repository: ipdk-io/networking-recipe
-          path: recipe
+          repository: ${{ env.DEPS_REPOSITORY }}
+          tag: ${{ env.DEPS_TAG }}
+          fileName: ${{ env.DEPS_FILENAME }}
+      - run: |
+          sudo tar -xzf $DEPS_FILENAME -C /
+          rm $DEPS_FILENAME
 
       - name: Install prerequisites
         run: |
           sudo apt install $PREREQS
 
-      - name: Install DPDK SDE
+      - name: Build krnlmon
+        working-directory: recipe
+        run: |
+          export DEPEND_INSTALL=$DEPS_INSTALL_DIR
+          export SDE_INSTALL=$SDE_INSTALL_DIR
+          ./make-all.sh --target=dpdk --rpath --no-ovs --no-build
+          cmake --build build -j4 --target krnlmon
```

Comment:
`--target krnlmon`?

What does this mean?

Reply:
It instructs cmake to build the `krnlmon` target. It's the equivalent of `make -C build krnlmon`.

Reply:
I didnt realize krnlmon target was supported

Reply:
We're just full of surprises. 😉

Reply:
Since krnlmon is treated as separate target, is krnlmon a separate process and not a thread anymore? Do we need to start krnlmon separately?

Reply:
We're speaking of krnlmon here as a cmake target, not a TDI target.

- `--target=dpdk` on the `make-all.sh` command line is a TDI target.
- `--target krnlmon-test` on the cmake command line is a cmake build target.

What I'm doing here is using `./make-all.sh` to configure without building, and then telling cmake to build the `krnlmon-test` target instead of `all` targets. This target builds and executes the krnlmon unit unit tests. The tests are small and have very few dependencies, so they build and execute very quickly. The GitHub runner completes the operation in ~25 seconds.

The tests don't actually start krnlmon. They build against individual switchlink source files and exercise them in isolation. Process vs. thread never enters into the picture.

Remember how I had you partition the switchlink logic when you were working on krnlmon? This the reason. `switchlink_link_test`, for example, consists of:

```c

add_executable(switchlink_link_test
    switchlink_link_test.cc
    switchlink_globals.c
    switchlink_link.c
    switchlink_link.h
)


define_switchlink_test(switchlink_link_test)
```

Reply:
Here's a link to the record of the action: https://github.com/ipdk-io/krnlmon/actions/runs/6788471829